### PR TITLE
remove unused script and update the doc

### LINKED ISSUE #9422: Clean up .NET automation scripts
It looks like the [automation folder](https://github.com/Azure/azure-sdk-for-net/tree/main/eng/scripts/automation) contains code that is no longer consumed that needs to be cleaned so we don't confuse our users with outdated documentation. For example, file https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/automation/Invoke-GenerateAndBuild.ps1#L75 is using tools that we want to retire.
Also, looks like this script is no longer used and the one that is consumed is https://github.com/Azure/azure-sdk-for-net/blob/main/eng/scripts/Invoke-GenerateAndBuildV2.ps1

### PR DESCRIPTION
**Pull request by chunyu3**
# Contributing to the Azure SDK
Fix https://github.com/Azure/azure-sdk-tools/issues/9422

- Remove `Invoke-GenerateAndBuild.ps1` which is not used anymore
- Rename the script `Invoke-DataPlaneGenerateSDKPackage.ps1` to `Invoke-AutorestDataPlaneGenerateSDKPackage.ps1` to clear that this script only for generate SDK with OpenAPI/swagger.
- Update the doc to clarify that the scaffolding is for SDKs with TypeSpec

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by maririos — APPROVED** [review]
Thank you!

**Comment by weshaggard** [comment]
https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-net%20Invoke-GenerateAndBuild.ps1&type=code

I still see references to the script you are removing. We should either update those or clean those up as well.

**Comment by chunyu3** [comment]
> https://github.com/search?q=repo%3AAzure%2Fazure-sdk-for-net%20Invoke-GenerateAndBuild.ps1&type=code
> 
> I still see references to the script you are removing. We should either update those or clean those up as well.

hello @weshaggard I removed those files. Please help to review again. thanks

File: eng/codegen_to_sdk_config.json [hunk 1]
```diff
@@ -1,31 +0,0 @@
-{
```
Comment (weshaggard) [inline eng/codegen_to_sdk_config.json:1]: @raych1 can you confirm we don't use these any longer? I see we have similar files in the other languages https://github.com/search?q=org%3AAzure+codegen_to_sdk_config&type=code

Comment (raych1) [reply eng/codegen_to_sdk_config.json:1]: This file was previously used by the docker-based SDK generation tool, which had been deleted.

**Review by raych1 — APPROVED** [review]

**Comment by raych1** [comment]
@chunyu3 
There are some additional files can be deleted too. You might refer to this PR for the details.
https://github.com/Azure/azure-sdk-for-net/pull/27179/

**Comment by chunyu3** [comment]
> @chunyu3 There are some additional files can be deleted too. You might refer to this PR for the details. #27179

There will be follow-up PR for this.

**Pull request merged**